### PR TITLE
Implement new summary table layout

### PR DESCRIPTION
**Pull request by hbillings**
Implements the summary tables from #1394.

Screen Shot 2019-04-12 at 2 36 33 PM

### This pull request changes...
- Breaks apart years into separate tables
- Project totals becomes separated vertically by years
- New colors (these were not in the mockup, but I didn't realize that until it was too late, and I haven't tested their 508 compliance -- @austinhernandez, might need your input here)
- Removed old basscss util classes in favor of Sass and design system utils

### This pull request is ready to merge when...
- [x] Tests have been updated (and all tests are passing)
- [x] This code has been reviewed by someone other than the original author
- [x] The experience passes a basic manual accessibility audit (keyboard nav, screenreader, text scaling) OR an exemption is documented

### This feature is done when...
- [x] Design has approved the experience
- [x] Product has approved the experience

### CODE REVIEW

File: web/src/containers/BudgetSummary.js [hunk 1]
```diff
@@ -12,222 +12,205 @@ const categoryLookup = {
...
+  return (
+    <Fragment>
+      <tr>
+        <th headers="summary-budget-null1 summary-budget-null2">
+          {title}
+        </th>
+          <Fragment key={year}>
+            <td
+              className="font-family--mono right-align"
+              headers={`summary-budget-fy-${year} summary-budget-fy-${year}-total`}
```
[the remark this answers is not in the document]
Comment (hbillings) [reply web/src/containers/BudgetSummary.js]: Oops, I thought I changed these, but it turns out I just changed the Total table's headers. (I think I can probably condense these into a `HeaderRow` component like existed before, actually.) But yes, given how much work we're doing on the tables, auditing them when we're finished makes sense -- I'll spin that up.

[the remark this answers is not in the document]
Comment (hbillings) [reply web/src/containers/BudgetSummary.js]: Created #1469

**Comment by austinhernandez** [comment]
**Rationale:** There are some [usability best practices for the table](https://medium.com/mission-log/design-better-data-tables-430a30a00d8c), that I wanted to incorporate. 

**For the visual design:**
- [ ] font: courier, weight: normal font-size: normal (same as body text)
- [ ] could we keep the K and M all capitalized? (Unless there was a specific reason to do it, [it doesn't seem like it](https://english.stackexchange.com/questions/181917/mixing-use-of-k-for-thousands-and-mm-for-millions).)
- [ ] header color: `#DAE9F7`
- [ ] cell color: `#F3F8FC`
- [ ] don’t use center alignment - numerical data is right-aligned, textual data is left-aligned, and headers are aligned with their data if possible.

**Comment by hbillings** [comment]
@austinhernandez This is super helpful, thank you! I just wanted to clarify something about the colors: The colors you mentioned are the original, pre-design-system colors, and they seem a little... too purple? in comparison to the new primary blue we're bringing in. But if you think we should stick with them, I will happily make the change. 

Or maybe I should revert the colors and we can make a new ticket for implementing typography/colors for tables across the board?

**Comment by austinhernandez** [comment]
Yes, lets make it consistent. I'll have to look for new colors then. All the "best practice" articles are saying use color super sparingly and collapse the width of the columns only to fit the content.

image

image

...but I don't know what kind of research has gone into this so I don't want to change everything at this point. Let's skip the colors for now, and we can talk about that in pairing tomorrow.

**Pull request merged**